shoot/schema_shoot.go

```diff
@@ -322,21 +316,21 @@ func addonsResource() *schema.Resource {
 func dNSResource() *schema.Resource {
 	return &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			"provider": {
-				Type:        schema.TypeString,
-				Description: "Provider is the DNS provider type for the Shoot.",
-				Optional:    true,
-			},
+			// "provider": {
```

Comment:
Why do you comment out dns provider info here ?

Reply:
Two reasons why I comment out the `provider` and `secret_name` properties:
1. The provider was not working anyway, so nobody would be able to use it at the moment. In the expand it was directly cast to `in["providers"].([]corev1beta1.DNSProvider)`. This will not work.
2. The current `corev1beta1.DNS` structure expects a list of providers, not just one. Each provider can have a `DNSIncludeExcludes`, `SecretName`, `Zones` and a `Type`. So the current schema of the `DNS` property of the Terraform provider does not fit to the Gardener structure at all.

Therefore I would comment it out for now and add the functionality later again when we come up with a schema, which fits the Gardener structure.